rfcs/whitespace-handling.md

```diff
@@ -0,0 +1,78 @@
+
+Candidate values would be:
+
+####  pre-wrap 
+
+Sequences of white space are preserved. Lines are broken at newline characters, at `<br>`, and as necessary to fill line boxes.
+
+#### break-spaces
+The behavior is identical to that of pre-wrap, except that:
+
+ *   Any sequence of preserved white space always takes up space, including at the end of the line.
+ * A line breaking opportunity exists after every preserved white space character, including between white space characters.
+ * Such preserved spaces take up space and do not hang, and thus affect the box’s intrinsic sizes (min-content size and max-content size).
+
+Since this would show all whitespaces we will have to strip unwanted whitespace on setting/loading the editor content. This would mean any textnode only consisting of whitespace, any linebreak, ...
+
+### text input
+#### spaces
+
+Text input becomes simpler, spaces can just be inserted as a space into the VDOM. 
+
+#### linebreak
+Breaks are odd in html, being a void element. We could consider using linebreaks (inside a textnode) instead, though this likely has it's own downsides. In any case it's best to stay consistent and go for either the `br` element or linebreaks consistently.
+
+### backspace/delete
+Becomes easier because we can now assume any whitespace is visible to the user and as such removing it will be too. This means we can get rid of a lot of complex heuristics in the backspace handler.
+
+### loading content
+Load content through the insert-html command (and others) becomes more complex because we will want to filter out any unwanted whitespace and linebreaks.
+
+### exporting content
+Exporting content becomes more complex because we will want to translate whitespace that would normally fold (consecutive spaces, linebreaks, spaces at the end of a node, etc.) to something visible.
+
+## Caveats
+### are we the only ones?
+It's unclear whether other editors use this css property and that's something we should investigate. If they don't, perhaps there's a good reason not to do this.
+
+Update: did some research
+
+| editor | solution |
+| ------------- | ------------- |
+| prosemirror | use pre-wrap by default, plugins exist to use nbsp behaviour |
+| ckeditor 4 & 5 | use nbsp, seems this is mostly handled in the view |
+| tinymce | use pre-wrap |
+
+
+### old content
+Older documents, pasted content and our templates contain a lot of spurious whitespace which will likely be shown when we use this css property. We will need some kind of strategy to deal with that.
+
+### inconsistencies between layout in the editor and when rendered outside
+Both `pre-wrap` and `break-spaces` also break lines on newline characters, which is likely unwanted behaviour as these will not show up in exported documents unless they apply the same whitespace styling. So extra care needs to be taken to replace these newlines with proper breaks on export. 
```

Comment:
Likely we will go for inserting <br> elements as a linebreak strategy, so I think this is a non-issue?